### ISSUE TITLE
Did the fuzzy search behavior change recently?

### ISSUE THREAD

**Issue by mirosval** [liuchengxu/vim-clap#556]
**Environment**
 - OS:10.15.7
 - (Neo)Vim version:v0.5.0-nightly-260-g38efa1730
 - vim-clap version: a72ccfb17ede8f8f7bcfe79eb5211dab9b44524e (git commit)
 - Have you reproduced with a minimal vimrc: yes
 - Have you updated to the latest plugin version:yes
 - Have you upgraded to/compiled the latest Rust binary: yes

```
" Clap
let g:clap_provider_grep_delay = 0
let g:clap_provider_grep_blink = [0, 0]
let g:clap_provider_grep_opts = '--with-filename --no-heading --vimgrep --hidden -g "!.git/"'
nnoremap <C-p> :Clap files ++finder=fd --type f --hidden --no-ignore --exclude '.git' --exclude 'target' --exclude '.cache' --exclude 'node_modules' --exclude 'dist'<CR>
nnoremap <leader>ff :Clap<CR>
nnoremap <leader>fg :Clap grep2<CR>
nnoremap <leader>fb :Clap buffers<CR>
nnoremap <leader>fs :Clap grep ++query=<cword><CR>
```

**Describe the bug**
In the last few days, Clap fuzzy matching behavior changed. Before I could fuzzy search through the entire path, now I have to put spaces between words.

```

            has cargo: 1
            has maple: /Users/miroslav/.config/nvim/plugged/vim-clap/target/release/maple
           maple info: version 0.1.21, built for x86_64-apple-darwin by rustc 1.47.0 (18bf6b4f0 2020-10-07).

         has +python3: 0
has py dynamic module: 1
            has ctags: ctags without JSON output support
     Current FileType: netrw
Third Party Providers: []
       Global Options:
    let g:clap#autoload_dir = '/Users/miroslav/.config/nvim/plugged/vim-clap/autoload'
    let g:clap#provider_alias = {'gfiles': 'git_files', 'hist:': 'command_history', 'hist/': 'search_history'}
    let g:clap_background_shadow_blend = 50
    let g:clap_disable_bottom_top = 0
    let g:clap_disable_matches_indicator = v:false
    let g:clap_disable_optional_async = v:false
    let g:clap_disable_run_rooter = v:false
    let g:clap_enable_background_shadow = v:true
    let g:clap_enable_debug = v:false
    let g:clap_enable_icon = 0
    let g:clap_forerunner_status_sign = {'done': '•', 'running': '!', 'using_cache': '*'}
    let g:clap_insert_mode_only = v:false
    let g:clap_multi_selection_warning_silent = 0
    let g:clap_no_matches_msg = 'NO MATCHES FOUND'
    let g:clap_open_action = {'ctrl-v': 'vsplit', 'ctrl-x': 'split', 'ctrl-t': 'tab split'}
    let g:clap_popup_border = 'rounded'
    let g:clap_preview_size = 5
    let g:clap_providers_relaunch_code = '@@'
    let g:clap_search_box_border_style = 'curve'
    let g:clap_search_box_border_symbols = {'nil': ['', ''], 'curve': ['', ''], 'arrow': ['', '']}
  Provider Variables:
    let g:clap_provider_grep_blink = [0, 0]
    let g:clap_provider_grep_delay = 0
    let g:clap_provider_grep_opts = '--with-filename --no-heading --vimgrep --hidden -g "!.git/"'
```

**To Reproduce**
`ctrl+p` type `initv`

**Expected behavior**
I expect it to match path `config/nvim/init.vim`

**Actual behavior**
`NO MATCHES FOUND`

**Screenshots**
<img width="727" alt="Screenshot 2020-10-20 at 11 29 21" src="https://user-images.githubusercontent.com/1315417/96567874-9acf4980-12c7-11eb-96d6-59ae6d919550.png">
<img width="691" alt="Screenshot 2020-10-20 at 11 29 28" src="https://user-images.githubusercontent.com/1315417/96567876-9c007680-12c7-11eb-9706-1d0c29700ca9.png">

**Additional context**
I can see the path before I type the last character (`v`). It also works when I type `init v` because then it highlights the `v` with a different color.
This used to work as described just a couple a days ago. I update all components quite often and now its like this. I'm not saying its strictly *broken*, but I'd just like the old behavior back

**Comment by romgrk** [comment]
The fuzzy search behavior did change recently (#541) but it wasn't supposed to cause this kind of behavior.

 - Have you installed a new version of `maple`?
 - What's the output of `:checkhealth`?

**Comment by liuchengxu** [comment]
Hmm, can not reproduce it on my side. If this happened just a few days ago, can you find which commit is the last one that works at your expectation?

<img width="1680" alt="截屏2020-10-20 下午9 02 56" src="https://user-images.githubusercontent.com/8850248/96589645-a4809d80-1317-11eb-8c45-07cc6fddbcd7.png">

**Comment by mirosval** [comment]
Okay, problem solved. @romgrk 's hint was useful. I tried going back 1-2 months but the problem persisted. Then I tried ":healthcheck" - I hadn't tried this prior, because I thought vim-clap would just not work at all if something there was broken. 

But the actual issue was that I'd recently upgraded to Python 3.9.0 and that created a new pyenv that did not have `pyx` and `neovim` installed. So I guess that `vim-clap` installation maybe fell back to some smaller feature set with those packages not in scope? But once I installed those 2 packages, it works as before.

`pip install neovim pyx`

Thank you for all your help, you're the best ❤️
